src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE_QUERY = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_QUERY = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID_QUERY = "DELETE FROM books WHERE id = ?";
+    private static final String URL =
+            "jdbc:mysql://localhost:3306/book_db?user=root&password=12345";
+    private final Connection connection;
```

Comment:
Extract code about getting connection to the separate util class.
```suggestion
    private static final String URL =
            "jdbc:mysql://localhost:3306/book_db?user=root&password=12345";
    private final Connection connection;
```